python/ray/air/checkpoint.py

```diff
@@ -583,6 +672,12 @@ def to_uri(self, uri: str) -> str:
             )
 
         with self.as_directory() as local_path:
+            checkpoint_metadata_path = os.path.join(
+                local_path, _CHECKPOINT_METADATA_FILE_NAME
+            )
+            with open(checkpoint_metadata_path, "wb") as file:
+                pickle.dump(self.metadata, file)
```

Comment:
Does `as_directory` call `_to_directory`? In which case this is already being handled here: https://github.com/ray-project/ray/pull/28387/files#diff-5e1cc7107d4ba60e36c31b0b0bfa0bf8042c08b5dcad03dd8f21b68b5c19f2beR526

Reply:
Not always. If the checkpoint was created with a `local_path`, then it won't be called.